### PR TITLE
feat: AWS Lambda integration

### LINKED ISSUE #10073: [Feature]: Requesting AWS Lambda in Data Sources similar to S3
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Using S3 integration with Appsmith is very seamless. It just needs AccessKey and SecretKey. A similar interface to directly invoke Lambda functions would be very helpful. 

A direct API call to lambda is a bit involved due to request signing. 
[https://docs.aws.amazon.com/lambda/latest/dg/API_Invoke.html](https://docs.aws.amazon.com/lambda/latest/dg/API_Invoke.html
)

Currently, n8n has such an integration - [https://n8n.io/integrations/n8n-nodes-base.awsLambda](https://n8n.io/integrations/n8n-nodes-base.awsLambda)

### Why should this be worked on?

We use many lambdas for our internal use. Most of them are currently behind another server. Having this feature would make the API calls direct.

### PR DESCRIPTION
**Pull request by mohanarpit**
#### PR fixes following issue(s)
Fixes #10073

#### Type of change
- New feature (non-breaking change which adds functionality)

#### How Has This Been Tested?
- [x] Manual
- [x] JUnit
- [ ] Jest
- [ ] Cypress

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Introduced AWS Lambda plugin for executing and managing AWS Lambda functions.

- **Enhancements**
  - Updated server configuration to support new plugins.

- **Documentation**
  - Added constants for new plugins in the PluginConstants interface.

- **Tests**
  - Added test cases for AWS Lambda plugin functionality.

- **Chores**
  - Implemented migrations to add AWS Lambda plugin to existing workspaces.

### CODE REVIEW

File: app/server/appsmith-plugins/awsLambdaPlugin/src/test/java/com/external/plugins/AwsLambdaPluginTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        AWSLambda mockLambda = mock(AWSLambda.class);
+        ListFunctionsResult mockFunctionsResult = new ListFunctionsResult();
+        mockFunctionsResult.setFunctions(List.of(new FunctionConfiguration().withFunctionName("test-aws-lambda")));
+        when(mockLambda.listFunctions()).thenReturn(mockFunctionsResult);
+
+        Mono<ActionExecutionResult> resultMono =
+                pluginExecutor.execute(mockLambda, datasourceConfiguration, actionConfiguration);
+        StepVerifier.create(resultMono)
+                .assertNext(result -> {
+                    assertEquals(1, ((ArrayNode) result.getBody()).size());
```
[the remark this answers is not in the document]
Comment (mohanarpit) [reply app/server/appsmith-plugins/awsLambdaPlugin/src/test/java/com/external/plugins/AwsLambdaPluginTest.java:82]: Not asserting the name as it is a mocked value. It'll be like testing if the mock works properly.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration040AddAWSLambdaPlugin.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        Plugin plugin = new Plugin();
+        plugin.setName(PluginConstants.PluginName.AWS_LAMBDA_PLUGIN_NAME);
+        plugin.setType(PluginType.REMOTE);
+        plugin.setPluginName(PluginConstants.PluginName.AWS_LAMBDA_PLUGIN_NAME);
+        plugin.setPackageName(PluginConstants.PackageName.AWS_LAMBDA_PLUGIN);
+        plugin.setUiComponent("UQIDbEditorForm");
+        plugin.setDatasourceComponent("DbEditorForm");
+        plugin.setResponseType(Plugin.ResponseType.JSON);
+        plugin.setIconLocation("https://assets.appsmith.com/aws-lambda-logo.png");
+        //        plugin.setDocumentationLink("https://docs.appsmith.com/connect-data/reference/google-ai");
```
[the remark this answers is not in the document]
Comment (mohanarpit) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration040AddAWSLambdaPlugin.java:39]: Added doc link. Will populate it later.

File: app/server/appsmith-plugins/awsLambdaPlugin/src/main/java/com/external/plugins/AwsLambdaPlugin.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                    })
+                    .onErrorMap(
+                            Exception.class,
+                            e -> new AppsmithPluginException(
+                                    AppsmithPluginError.PLUGIN_ERROR, "Unsupported command: " + command))
+                    .map(obj -> {
+                        System.out.println("obj = " + obj.getBody());
+                        return obj;
+                    })
+                    .subscribeOn(Schedulers.boundedElastic());
```
[the remark this answers is not in the document]
Comment (mohanarpit) [reply app/server/appsmith-plugins/awsLambdaPlugin/src/main/java/com/external/plugins/AwsLambdaPlugin.java]: This is not true. `subscribeOn` is a non-blocking call.

**Review by trishaanand — APPROVED** [review]

File: app/server/appsmith-plugins/awsLambdaPlugin/src/main/java/com/external/plugins/AwsLambdaPlugin.java [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+            if (!StringUtils.hasText(region)) {
+                region = "us-east-1"; // Default region
+            }
+
+            AWSLambdaClientBuilder awsLambdaClientBuilder =
+                    AWSLambdaClientBuilder.standard().withRegion(Regions.fromName(region));
+
+            // If access key and secret key are not provided, use the default credentials provider chain. That will
+            // pick up the instance role if running on an EC2 instance.
+            if ("accessKey".equals(authenticationType)) {
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/awsLambdaPlugin/src/main/java/com/external/plugins/AwsLambdaPlugin.java:162]: Not moving these raw strings to constants as an experiment to see if it's possible to templatize & generate other similar plugin code using this as the base. Hence, putting all functions in a single class & not refactoring too much.

**Review by trishaanand — APPROVED** [review]

**Pull request merged**